.github/workflows/publish-a-package.yml

```diff
@@ -0,0 +1,29 @@
+name: Publish a Package from Tag
+
+on:
+  push:
+    tags:
+      - 'opentelemetry-+==1.+'
+
+jobs:
+  publish:
+    name: Publish package from tag ${{ env.GITHUB_REF }}
```

Comment:
This variable is _crucial_ to make this whole setup work. The variable is defined [on GitHub's docs](https://docs.github.com/en/actions/reference/environment-variables#default-environment-variables) and I took inspiration from [this SO post on how to get the currently pushed tag in GitHub actions](https://stackoverflow.com/a/58178121/7460739) for the rest of the PR.